### PR TITLE
feat: support any Inspector for block + tx tracing

### PR DESCRIPTION
**Pull request by mattsse**
unblocks #6977

Previously all tx+block tracing helper trait functions were limited to `TracingInspector`.

This adds these implementations as additional functions that are generic over `Inspector` trait and reuses them.

### CODE REVIEW

File: crates/rpc/rpc/src/eth/api/transactions.rs [hunk 1]
```diff
@@ -306,17 +332,59 @@ pub trait EthTransactions: Send + Sync {
...
+    ) -> EthResult<Option<Vec<R>>>
+    where
+        // This is the callback that's invoked for each transaction with the inspector, the result,
+        // state and db
+        F: for<'a> Fn(
+                TransactionInfo,
+                Insp,
+                ExecutionResult,
+                &'a State,
+                &'a StateCacheDB,
```
Comment (Rjected) [inline crates/rpc/rpc/src/eth/api/transactions.rs:378]: so `State` = before, `StateCacheDB` = after?

File: crates/rpc/rpc/src/eth/api/transactions.rs [hunk 2]
```diff
@@ -306,17 +332,59 @@ pub trait EthTransactions: Send + Sync {
...
+
+    /// Executes all transactions of a block and returns a list of callback results invoked for each
+    /// transaction in the block.
+    ///
+    /// This
+    /// 1. fetches all transactions of the block
+    /// 2. configures the EVM evn
+    /// 3. loops over all transactions and executes them
+    /// 4. calls the callback with the transaction info, the execution result, the changed state
+    /// _after_ the transaction [StateProviderDatabase] and the database that points to the state
```
Comment (Rjected) [inline crates/rpc/rpc/src/eth/api/transactions.rs:359]:
```suggestion
    /// _after_ the transaction [StateCacheDB] and the database that points to the state
```

**Review by Rjected — APPROVED** [review]
lgtm, just two comments about statecachedb / state

**Comment by Rjected** [comment]
ah rip, the suggestion linked to a private item

File: crates/rpc/rpc/src/eth/api/transactions.rs (hunk 1 above)
Comment (mattsse) [reply crates/rpc/rpc/src/eth/api/transactions.rs:379]: added another sentence with more context

Comment (Rjected) [reply crates/rpc/rpc/src/eth/api/transactions.rs:379]: nice, the new docs make it very clear

**Pull request merged**